packages/core/src/hydration/utils.ts

```diff
@@ -241,8 +241,10 @@ export function processTextNodeMarkersBeforeHydration(node: HTMLElement) {
  * Internal type that represents a claimed node.
  * Only used in dev mode.
  */
-type ClaimedNode = {
-  __claimed?: boolean;
+export type HydrationNode = {
+  __hydration?: {
+    status: 'claimed'|'skipped',
+  }|{status: 'mismatched', actualNode: string | null, expectedNode: string | null};
```

Comment:
nit: adding `Details` to indicate that these are not actual nodes, but an info/details about the nodes.
```suggestion
  }|{status: 'mismatched', actualNodeDetails: string | null, expectedNodeDetails: string | null};
```